ir/src/helpers.rs

```diff
@@ -2,37 +2,43 @@ use crate::error::SemanticError;
 
 /// Struct to help with validation of AIR source.
 pub(super) struct SourceValidator {
-    trace_columns_exists: bool,
+    main_trace_columns_exists: bool,
+    aux_trace_columns_exists: bool,
     public_inputs_exists: bool,
     boundary_constraints_exists: bool,
     integrity_constraints_exists: bool,
+    random_values_exists: bool,
 }
 
 impl SourceValidator {
     pub fn new() -> Self {
         SourceValidator {
-            trace_columns_exists: false,
+            main_trace_columns_exists: false,
+            aux_trace_columns_exists: false,
```

Comment:
At first I did this, but then I thought it might be a little confusing, because the presence of `trace_columns` may mean the presence of both columns. Therefore, I decided that the division into main and aux would more accurately reflect the presence of each of the columns. But I agree with you, we cannot have a situation where `main_trace_columns` are not exist, because we check this at the parser level

Reply:
Makes sense.